lib/PhasarLLVM/ControlFlow/LLVMBasedICFGGlobalsImpl.cpp

```diff
@@ -136,7 +136,14 @@ static llvm::Function *createDtorCallerForModule(
 
   for (auto It = RegisteredDtors.rbegin(), End = RegisteredDtors.rend();
        It != End; ++It) {
-    IRB.CreateCall(It->first, {It->second});
+    auto FunCallee = It->first;
+    assert(FunCallee.getFunctionType()->getNumParams() == 1);
+    auto *ExpectedArgType = FunCallee.getFunctionType()->getParamType(0);
+    auto *Arg = It->second;
+    if (Arg->getType() != ExpectedArgType) {
+      Arg = IRB.CreateBitOrPointerCast(Arg, ExpectedArgType);
+    }
```

Comment:
Can we be sure that the call is correctly matched? Otherwise, we could introduce wrong cast here.